### ISSUE TITLE
FSDP2 grad_norm=0 and loss is not decreasing

### ISSUE THREAD

**Issue by mahaozhe** [ByteDance-Seed/VeOmni#118]
### System Info

When using fsdp1, all is good, while switched to fsdp2, the loss is not decreasing and grad_norm = 0 for all training steps.

### Information

- [x] The official example scripts
- [ ] My own modified scripts

### Tasks

- [x] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Using the sample train bash and qwen2.5 script

### Expected behavior

The fsdp2 seems not working correctly.

**Comment by Luosuu** [comment]
can you share more details about your run like what model and datasets? It is better if you can share your job yaml directly. Thank you

**Comment by mahaozhe** [comment]
Hi, @Luosuu , I find the reason. For me, I'm using the `tasks/omni/train_qwen2_5_vl.py` as a template to adapt the InternVL3.5 model. I fixed the issue by adding `weights_path=args.model.model_path` to the `build_parallelize_model` function: <https://github.com/ByteDance-Seed/VeOmni/blob/main/tasks/omni/train_qwen2_5_vl.py#L254>. Otherwise, it goes to [this line](https://github.com/ByteDance-Seed/VeOmni/blob/main/veomni/distributed/torch_parallelize.py#L384), so that the model is initialized empty (I guess all-0 initialized, cause the loss keeps to be 11.93 ($\sim \ln(151936)$, equal logits for all tokens) and grad_norm is 0). When the `weights_path` is indicated, it works for me now. Maybe need to double check, thanks a lot!

**Comment by Luosuu** [comment]
thank you for sharing the finding!
